src/main/java/com/mastercard/developer/oauth/OAuth.java

```diff
@@ -101,22 +101,28 @@ static String getSignatureBaseString(String httpMethod, String baseUri, String p
    * @return Sorted map of query parameter key/value pairs. Values for parameters with the same name are added into a list.
    */
   static TreeMap<String, List<String>> extractQueryParams(URI uri, Charset charset) {
-    final String queryParams = uri.getQuery();
-    if (queryParams == null || queryParams.isEmpty()) {
+
+    final String decodedQueryString = uri.getQuery();
+    final String rawQueryString =  uri.getRawQuery();
+    if (decodedQueryString == null || decodedQueryString.isEmpty()
+            || rawQueryString == null || rawQueryString.isEmpty()) {
+      // No query params
       return new TreeMap<>();
     }
 
+    boolean isDecoded = decodedQueryString.equals(rawQueryString);
+
     final TreeMap<String, List<String>> queryPairs = new TreeMap<>();
-    final String[] pairs = queryParams.split("&");
+    final String[] pairs = decodedQueryString.split("&");
     for (String pair : pairs) {
       final int idx = pair.indexOf('=');
       String key = idx > 0 ? pair.substring(0, idx) : pair;
       if (!queryPairs.containsKey(key)) {
-        key = Util.percentEncode(key, charset);
+        key = isDecoded ? key : Util.percentEncode(key, charset);
```

Comment:
Instead of isDecoded, would isEncoded make more intuitive sense? To me, decoded means it's just in non-encoded form.